### DISCUSSION TITLE
Allow setting relativenumber in config

### DISCUSSION THREAD

**Discussion by tlacuilose** [CopilotC-Nvim/CopilotChat.nvim#196] [Q&A] [answered]
The chat window has relativenumber off by default. Allow changing this option in the config. 

This would make it more comfortable for us vim users that use relative numbers to move around. I find it particularly useful when yanking just a few lines from the suggestion in the chat window. 

Thanks for creating this plugin!

**Comment by deathbeam** [comment, marked as answer]
Replied this also on the PR but they are just buffers so you can work with them like you normally work with buffers for eveything else, imo confguration is not necessary:

```lua
vim.api.nvim_create_autocmd("BufEnter", {
    pattern = 'copilot-*',
    callback = function()
        vim.opt_local.relativenumber = true
        vim.opt_local.number = true
    end
})
```

EDIT: But also maybe it would be good idea to actually have the buffer names documented in README as atm we dont
EDIT2: Also we probably dont need to overwrite relativenumber at all for the window, so it could be straight up removed so it takes users default config.

**Comment by tlacuilose** [comment]
I don't remember ever seeing buffer names in other plugins' READMEs. But I think it is a good idea, anyways these are the ones I could find for copilot chat:
- `copilot-diff'` ([init.lua#L454](https://github.com/CopilotC-Nvim/CopilotChat.nvim/blob/17fbec0e33c2a0e1c6532fae6600935114e0df1c/lua/CopilotChat/init.lua#L454))
-  `copilot-system-prompt` ([init.lua#L498](https://github.com/CopilotC-Nvim/CopilotChat.nvim/blob/17fbec0e33c2a0e1c6532fae6600935114e0df1c/lua/CopilotChat/init.lua#L498))
- `copilot-user-selection` ([init.lua#L512](https://github.com/CopilotC-Nvim/CopilotChat.nvim/blob/17fbec0e33c2a0e1c6532fae6600935114e0df1c/lua/CopilotChat/init.lua#L512))
- `copilot-chat` ([chat.lua#L42](https://github.com/CopilotC-Nvim/CopilotChat.nvim/blob/17fbec0e33c2a0e1c6532fae6600935114e0df1c/lua/CopilotChat/chat.lua#L42))

On EDIT 2 that is a good point. However I don't know if this was a deliberate choice for user experience for example  with wrap, copilot gives long sentences or even paragraphs in a single line. So the plugin having full control of this option is actually a good thing.

Thank you for the suggestion, it works great. Let me know if I could raise any PRs for the other comments in the edits or maybe we should just close this issue.

**Comment by deathbeam** [comment]
> I don't remember ever seeing buffer names in other plugins' READMEs. But I think it is a good idea, anyways these are the ones I could find for copilot chat:
> 
> * `copilot-diff'` ([init.lua#L454](https://github.com/CopilotC-Nvim/CopilotChat.nvim/blob/17fbec0e33c2a0e1c6532fae6600935114e0df1c/lua/CopilotChat/init.lua#L454))
> * `copilot-system-prompt` ([init.lua#L498](https://github.com/CopilotC-Nvim/CopilotChat.nvim/blob/17fbec0e33c2a0e1c6532fae6600935114e0df1c/lua/CopilotChat/init.lua#L498))
> * `copilot-user-selection` ([init.lua#L512](https://github.com/CopilotC-Nvim/CopilotChat.nvim/blob/17fbec0e33c2a0e1c6532fae6600935114e0df1c/lua/CopilotChat/init.lua#L512))
> * `copilot-chat` ([chat.lua#L42](https://github.com/CopilotC-Nvim/CopilotChat.nvim/blob/17fbec0e33c2a0e1c6532fae6600935114e0df1c/lua/CopilotChat/chat.lua#L42))
> 
> On EDIT 2 that is a good point. However I don't know if this was a deliberate choice for user experience for example with wrap, copilot gives long sentences or even paragraphs in a single line. So the plugin having full control of this option is actually a good thing.
> 
> Thank you for the suggestion, it works great. Let me know if I could raise any PRs for the other comments in the edits or maybe we should just close this issue.

Yea that should be all of them. And on EDIT 2, everything other than relativenumbers was deliberate choice because of what you said, but relativenumbers config is not rly necessary. So I think opening PR for extending the documentation + maybe example with autocommand and removal of relativenumber override we are setting is proably fine.

**Reply by tlacuilose** [reply]
Here, I have raised that [PR](https://github.com/CopilotC-Nvim/CopilotChat.nvim/pull/199).

**Reply by deathbeam** [reply]
Thanks a lot, merged
